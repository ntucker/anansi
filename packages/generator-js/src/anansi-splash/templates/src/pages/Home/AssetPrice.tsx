import { useLive } from '@data-client/react';
import { getTicker } from 'resources/Ticker';

/** Shows the current trading price for a given asset */
export default function AssetPrice({ symbol }: Props) {
  const product_id = `${symbol}-USD`
  // Learn more about Reactive Data Client: https://dataclient.io/docs
  const ticker = useLive(getTicker, { product_id });
  const displayPrice = formatPrice.format(ticker.price);
  return (
    <span>
      {symbol} {displayPrice}
    </span>
  );
}
export interface Props {
  symbol: string;
}

const formatPrice = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
})
