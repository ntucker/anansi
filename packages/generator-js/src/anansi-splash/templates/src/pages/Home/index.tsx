import AssetPrice from './AssetPrice';
<% if (style === 'linaria') { %>
import { styled } from '@linaria/react';

const Head = styled.p`
  font-size: 30px;
`;
<% } else { %>
const Head = ({children}) => <p style={{ fontSize: '30px' }}>{children}</p>;
<% } %>

export default function Home() {
  return (
    <>
      <Head>
        Congrats! You&apos;ve created <%= appName %>!
      </Head>
      <p style={{ fontSize: '15px' }}>
        Check out the generated ReadMe for instructions on how to use this
        library
      </p>
      <p style={{ fontSize: '24px' }}>
        <AssetPrice symbol="BTC" />
      </p>
    </>
  );
}
