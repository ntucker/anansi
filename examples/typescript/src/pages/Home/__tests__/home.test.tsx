import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../index';

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
