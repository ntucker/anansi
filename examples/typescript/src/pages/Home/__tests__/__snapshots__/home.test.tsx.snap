// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main"
>
  <h3>
    Hello 
    <img
      src="SvgrURL"
      style={
        Object {
          "width": "1em",
        }
      }
    />
     
    <div
      style={
        Object {
          "width": "1em",
        }
      }
    >
      [Mocked SVG]
    </div>
    World 
    <a
      href="https://true.io"
    >
      True IO
    </a>
  </h3>
  <p
    className="mountainTop"
  >
    Ok again 
    <img
      src="FileURL"
    />
     
    /hi/bob/test.txt
     
    https://api.github.com/repos/issues
     
  </p>
  <p
    className="silly"
  >
    $text-color: 
    textColor
  </p>
  <p>
    <img
      src="SvgrURL"
    />
  </p>
  <code>
    &lt;html&gt;
  &lt;body&gt;&lt;h1&gt;My HTML Works&lt;/h1&gt;&lt;/body&gt;
&lt;/html&gt;

  </code>
</div>
`;
